integration_tests/src/main/python/parquet_write_test.py

```diff
@@ -279,6 +279,19 @@ def test_parquet_write_legacy_fallback(spark_tmp_path, ts_write, ts_rebase, spar
             'DataWritingCommandExec',
             conf=all_confs)
 
+@allow_non_gpu('DataWritingCommandExec')
+def test_parquet_write_encryption_fallback(spark_tmp_path, spark_tmp_table_factory):
+    gen = IntegerGen()
+    data_path = spark_tmp_path + '/PARQUET_DATA'
+    assert_gpu_fallback_write(
```

Comment:
There needs to be a test for falling back on Parquet reads, not just writes, when either of the encrypted keys are set.  For the read test, the CPU should perform the encrypted write.

Reply:
The issue #5439 specifically talks about writes. Should we tackle the reads as part of a separate PR which will completely fix #5246 or should we expand #5439 to include reads?

Reply:
We can handle reads in a separate PR.